### PR TITLE
Add model training with XGBoost

### PR DESCRIPTION
**Pull request by rak340**
## What Changes Were Made
Implemented XGBoost classifier to predict penguin species using seaborn dataset.

## Why These Changes Were Needed
To enable model training and accuracy evaluation using real-world ML pipeline.

## Testing Performed
Ran `uv run python main.py`. Output: `Model accuracy: 1.00`

## Screenshots (if applicable)
image

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+# Drop rows with missing values
+penguins = penguins.dropna()
+
+# Encode categorical variables
+le_species = LabelEncoder()
+penguins['species'] = le_species.fit_transform(penguins['species'])
+le_island = LabelEncoder()
+penguins['island'] = le_island.fit_transform(penguins['island'])
+le_sex = LabelEncoder()
+penguins['sex'] = le_sex.fit_transform(penguins['sex'])
```
Comment (SatyaMohan95) [inline main.py:20]: you could consider using a loop for the laberEncoder for each column which could make the code a bit more concise

File: main.py [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+# Split data
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+# Train XGBoost model
+model = xgb.XGBClassifier(random_state=42)
+model.fit(X_train, y_train)
+
+# Predict and evaluate
+y_pred = model.predict(X_test)
+accuracy = accuracy_score(y_test, y_pred)
```
Comment (SatyaMohan95) [inline main.py:35]: can we test the score with other metrics instead of the accuracy, as it is not that reliable. as model accuracy is 1.00 is it overfitting?

**Review by SatyaMohan95 — CHANGES_REQUESTED** [review]

**Comment by rak340** [comment]
@SatyaMohan95 Kindly review and let me know if i need to do any improvements or provide your approval.

**Review by SatyaMohan95 — APPROVED** [review]
Label encoder is now in the loop. 
Accuracy is replaced by F1, precision , recall, and calibration curve.

I am satisfied with the code and i will be moving with approval

**Comment by SatyaMohan95** [comment]
Assigned Role: Rakesh Kasaragadda 

Role: Model Training, Evaluation Metrics, Code Reviewer

Workflow Demonstrated: Merge

Summary of Contribution:
 Created feature branch feat-rakesh-train-model
 Implemented XGBoost classifier for penguin species prediction
 Performed label encoding using a loop
 Added model evaluation logic using:
   Precision
   Recall
  F1-score
   Calibration Curve
Responded to review feedback and pushed updates
Reviewed Saketh's PR and approved changes
This PR was reviewed and merged into main by Satya
